### PR TITLE
Test Android session reuse

### PR DESCRIPTION
**Pull request by yschimke**
Test for https://github.com/square/okhttp/issues/5342

### CODE REVIEW

File: okhttp-testing-support/src/main/java/okhttp3/RecordingEventListener.java [hunk 1]
```diff
@@ -75,7 +76,7 @@ private void logEvent(CallEvent e) {
     CallEvent startEvent = e.closes();
 
     if (startEvent != null) {
-      assertThat(eventSequence).contains(startEvent);
+      assertTrue(eventSequence.contains(startEvent));
```
Comment (yschimke) [inline okhttp-testing-support/src/main/java/okhttp3/RecordingEventListener.java]: Not compatible with older android due to internal stream usage.

**Comment by yschimke** [comment]
image

File: okhttp-testing-support/src/main/java/okhttp3/RecordingEventListener.java (hunk 1 above)
Comment (yschimke) [reply okhttp-testing-support/src/main/java/okhttp3/RecordingEventListener.java]: n.b. this is also a move to testing support

File: android-test/src/androidTest/java/okhttp/android/test/OkHttpTest.kt [hunk 2]
```diff
@@ -190,6 +193,75 @@ class OkHttpTest {
...
+
+    client.connectionPool.evictAll()
+    assertEquals(0, client.connectionPool.connectionCount())
+
+    client.newCall(request).execute().use { response ->
+      assertEquals(200, response.code)
+    }
+
+    assertEquals(2, sessionIds.size)
+    assertEquals(sessionIds[0], sessionIds[1])
```
Comment (swankjesse) [inline android-test/src/androidTest/java/okhttp/android/test/OkHttpTest.kt]: Sweet test.

Comment (ragner) [reply android-test/src/androidTest/java/okhttp/android/test/OkHttpTest.kt:262]: Hi @yschimke, I am trying this session reuse test with Android 7.1.1 but it is not working for me. 
What version of Android is compatible with session reuse?

Comment (yschimke) [reply android-test/src/androidTest/java/okhttp/android/test/OkHttpTest.kt:262]: I think that's a question for https://github.com/google/conscrypt

**Pull request merged**